### ISSUE TITLE
NEURON 7.8.x breaks calculation of axial currents from vmem

### ISSUE THREAD

**Issue by espenhgn** [LFPy/LFPy#209]
The newest NEURON version (tested with 7.8.1 from PyPI) breaks the calculation of axial currents from membrane voltages  using `LFPy.Cell.get_axial_currents_from_vmem()` in fundamental ways, if sections connect outside end points of sections (and possibly by creating more than two children branches at the end of some parent section). 

This issue results in a host of broken tests and segmentation faults in tests that used to run without issues.

This requires attention ASAP. I created a working branch fixing another issue with use of `neuron.h.SectionRef()` at https://github.com/espenhgn/LFPy/tree/iaxial that exposes this problem:
```
$ py.test -v LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_01
================================================================================== test session starts ==================================================================================
platform darwin -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /Users/ehagen/anaconda3/envs/nrn78/bin/python
cachedir: .pytest_cache
rootdir: /Users/ehagen/Repositories/LFPy
collected 1 item                                                                                                                                                                        

LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_01 FAILED                                                                                                [100%]

======================================================================================= FAILURES ========================================================================================
__________________________________________________________________ testCell.test_cell_get_axial_currents_from_vmem_01 ___________________________________________________________________

self = <LFPy.test.test_cell.testCell testMethod=test_cell_get_axial_currents_from_vmem_01>

    def test_cell_get_axial_currents_from_vmem_01(self):
        '''
        Check Kirchhoff in soma when single dend connected to soma mid.
        '''
        neuron.h('forall delete_section()')
        soma = neuron.h.Section(name='soma')
        dend = neuron.h.Section(name='dend')
        dend.connect(soma(0.5), 0)
        morphology = neuron.h.SectionList()
        morphology.wholetree()
        cell = cell_w_synapse_from_sections(morphology)
        iaxial, d_list, pos_list = cell.get_axial_currents_from_vmem()
    
        imem = copy(cell.imem)
        # some cleanup of Python-created section references
        cell = None
        morphology = None
        dend = None
        soma = None
        neuron.h('forall delete_section()')
    
>       np.testing.assert_almost_equal(-iaxial[0], imem[0], decimal=9)
E       AssertionError: 
E       Arrays are not almost equal to 9 decimals
E       
E       Mismatched elements: 3136 / 3201 (98%)
E       Max absolute difference: 0.01619416
E       Max relative difference: 0.50000939
E        x: array([-0.000000000e+00, -0.000000000e+00, -0.000000000e+00, ...,
E               9.092826669e-07,  9.064551682e-07,  9.036276695e-07])
E        y: array([0.000000000e+00, 0.000000000e+00, 0.000000000e+00, ...,
E              1.818562662e-06, 1.812889563e-06, 1.807233831e-06])

LFPy/test/test_cell.py:709: AssertionError
================================================================================ short test summary info ================================================================================
FAILED LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_01 - AssertionError: 
=================================================================================== 1 failed in 1.66s ===================================================================================
(nrn78) 1x-193-157-180-41:LFPy ehagen$ 
```

Segmentation fault 11 when running the test suite:
```
$ py.test -v LFPy/test/test_cell.py
================================================================================== test session starts ==================================================================================
platform darwin -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /Users/ehagen/anaconda3/envs/nrn78/bin/python
cachedir: .pytest_cache
rootdir: /Users/ehagen/Repositories/LFPy
collected 84 items                                                                                                                                                                      

LFPy/test/test_cell.py::testCell::test_cell_cellpickler_00 PASSED                                                                                                                 [  1%]
LFPy/test/test_cell.py::testCell::test_cell_chiral_morphology_00 PASSED                                                                                                           [  2%]
LFPy/test/test_cell.py::testCell::test_cell_chiral_morphology_01 PASSED                                                                                                           [  3%]
LFPy/test/test_cell.py::testCell::test_cell_chiral_morphology_02 PASSED                                                                                                           [  4%]
LFPy/test/test_cell.py::testCell::test_cell_distort_geometry_01 PASSED                                                                                                            [  5%]
LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_00 PASSED                                                                                                [  7%]
LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_01 Fatal Python error: Segmentation fault

Current thread 0x00000001123065c0 (most recent call first):
  File "/Users/ehagen/Repositories/LFPy/LFPy/cell.py", line 301 in __init__
  File "/Users/ehagen/Repositories/LFPy/LFPy/test/test_cell.py", line 1973 in cell_w_synapse_from_sections
  File "/Users/ehagen/Repositories/LFPy/LFPy/test/test_cell.py", line 698 in test_cell_get_axial_currents_from_vmem_01
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/unittest/case.py", line 633 in _callTestMethod
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/unittest/case.py", line 676 in run
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/unittest/case.py", line 736 in __call__
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/unittest.py", line 278 in runtest
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 153 in pytest_runtest_call
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 247 in <lambda>
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 294 in from_call
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 246 in call_runtest_hook
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 207 in call_and_report
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 117 in runtestprotocol
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/runner.py", line 100 in pytest_runtest_protocol
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/main.py", line 321 in pytest_runtestloop
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/main.py", line 296 in _main
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/main.py", line 240 in wrap_session
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/main.py", line 289 in pytest_cmdline_main
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/Users/ehagen/anaconda3/envs/nrn78/lib/python3.8/site-packages/_pytest/config/__init__.py", line 157 in main
  File "/Users/ehagen/anaconda3/envs/nrn78/bin/py.test", line 11 in <module>
Segmentation fault: 11
```

**Comment by espenhgn** [comment]
The issue seem to arise due to
`LFPy.Cell.get_dict_parent_connections`:
```
def get_dict_parent_connections(self):
        """
        Return dictionary with parent connection point for all sections.
        Returns
        -------
        connection_dict : dictionary
            Dictionary containing a float in range [0, 1] for each section
            in cell. The float gives the location on the parent segment
            to which the section is connected.
            The dictionary is needed for computing axial currents.
        """
        connection_dict = {}
        for i, sec in enumerate(self.allseclist):
            connection_dict[sec.name()] = neuron.h.parent_connection()
        return connection_dict
```

Setting `neuron.h.parent_connection(sec=sec)`, which should be the correct way to call the function (https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/topology.html?highlight=parent_connection#parent_connection) will make failing tests run and vice versa. Bizarre!

**Comment by torbjone** [comment]
I'm confused, did you solve the problem by adding the "sec=sec" argument, or is this still a mystery?

**Comment by espenhgn** [comment]
No, it's not a fix. Setting `sec=sec` will fix `LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_01` but break `LFPy/test/test_cell.py::testCell::test_cell_get_axial_currents_from_vmem_00` and so forth...

**Comment by espenhgn** [comment]
It seems we need to provide `sec` only in cases where sections connect outside terminal points but that sounds like a brittle fix. 
On 7.7.2 one can set `sec=sec` and have all tests run fine.

**Comment by espenhgn** [comment]
Hi @torbjone . I created PR #210 with some formatting/fixes, but this does not fix the issue with NEURON v7.8. I think the root of the issue is that we've been calling methods like `neuron.h.ri(x)`, `neuron.h.parent_section` without their optional `sec=<section reference>` parameter. Then the "currently accessed section" is being used by default, and in v7.8 this is now somehow stricter/less implicit.

Should we unsupport newer versions than 7.7.2 for now?  That would be unfortunate....

I think we need to reimplement `get_axial_currents_from_vmem()` in a way that is understandable (cf. #148).

**Comment by torbjone** [comment]
This seems like a sensible change from NEURON. Unsupporting the version of NEURON that is now standard (and on pip) seems very bad, so we should obviously try to avoid this if we can. 
But I don't fully understand this issue: Do we expect that reimplementing get_axial_currents_from_vmem() will solve all problems, or is it also a problem that we should only use "sec=sec" as an argument in some cases, but not in all?

**Comment by espenhgn** [comment]
Hi @torbjone, the issue(s) with get_axial_currents_from_vmem() was only the first I noticed, because the test_cell.py file get executed first. There are other test suites that are currently broken that I haven't even looked at yet: 
```
$ py.test -v LFPy/test/test_network.py 
LFPy/test/test_network.py::testNetworkPopulation::test_NetworkPopulation_00 PASSED                                                                                                [ 14%]
LFPy/test/test_network.py::testNetwork::test_Network_00 PASSED                                                                                                                    [ 28%]
LFPy/test/test_network.py::testNetwork::test_Network_01 PASSED                                                                                                                    [ 42%]
LFPy/test/test_network.py::testNetwork::test_Network_02 PASSED                                                                                                                    [ 57%]
LFPy/test/test_network.py::testNetwork::test_Network_03 PASSED                                                                                                                    [ 71%]
LFPy/test/test_network.py::testNetwork::test_Network_04 PASSED                                                                                                                    [ 85%]
LFPy/test/test_network.py::testNetwork::test_Network_05 Fatal Python error: Segmentation fault

$ py.test -v LFPy/test/test_networkcell.py # same for test_templatecell.py as the tests are identical
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_cellpickler_00 PASSED                                                                                                   [  1%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_chiral_morphology_00 PASSED                                                                                             [  3%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_chiral_morphology_01 PASSED                                                                                             [  5%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_chiral_morphology_02 PASSED                                                                                             [  7%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_distort_geometry_01 PASSED                                                                                              [  8%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_closest_idx_00 PASSED                                                                                               [ 10%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_idx_00 PASSED                                                                                                       [ 12%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_idx_children_00 PASSED                                                                                              [ 14%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_idx_name_00 PASSED                                                                                                  [ 15%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_idx_parent_children_00 PASSED                                                                                       [ 17%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_intersegment_distance_00 PASSED                                                                                     [ 19%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_intersegment_vector_00 PASSED                                                                                       [ 21%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_rand_idx_area_norm_00 FAILED                                                                                        [ 22%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_rand_prob_area_norm_00 PASSED                                                                                       [ 24%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_rand_prob_area_norm_from_idx PASSED                                                                                 [ 26%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_get_rand_prob_area_norm_from_idx_00 PASSED                                                                              [ 28%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_point_process_00 PASSED                                                                                             [ 29%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_00 PASSED                                                                                                       [ 31%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_01 PASSED                                                                                                       [ 33%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_02 PASSED                                                                                                       [ 35%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_03 PASSED                                                                                                       [ 36%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_04 PASSED                                                                                                       [ 38%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_05 PASSED                                                                                                       [ 40%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_pos_06 PASSED                                                                                                       [ 42%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_00 PASSED                                                                                                  [ 43%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_01 PASSED                                                                                                  [ 45%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_02 PASSED                                                                                                  [ 47%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_03 PASSED                                                                                                  [ 49%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_04 PASSED                                                                                                  [ 50%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_05 PASSED                                                                                                  [ 52%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_rotation_06 PASSED                                                                                                  [ 54%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_set_synapse_00 PASSED                                                                                                   [ 56%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_simulate_current_dipole_moment_00 PASSED                                                                                [ 57%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_simulate_current_dipole_moment_01 PASSED                                                                                [ 59%]
LFPy/test/test_networkcell.py::testNetworkCell::test_cell_simulate_current_dipole_moment_02 Fatal Python error: Segmentation fault

$ py.test -v LFPy/test/test_pointprocess.py
LFPy/test/test_pointprocess.py::testPointProcess::test_PointProcess_00 PASSED                                                                                                     [  9%]
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_00 PASSED                                                                                                               [ 18%]
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_01 Fatal Python error: Segmentation fault
```

So it's a matter of starting in one place or the other, I presume. I would also much prefer to support 7.8 in the 2.1.1rc. 

To respond to your question; I don't think providing `sec=sec` is a universal fix. There could be reasons to use the currently accessed section, but I would guess it makes more sense to make these statements explicit. 

I went over the LFPy.Cell class already, but the one place where NEURON behavior may be different is at https://github.com/LFPy/LFPy/blob/0452c48b4d240ef6a2801bc7bff0a193158ff1d6/LFPy/cell.py#L2515 for instance. 
One would think that `sec=parentsec` should be provided as well, but the output is different then between NEURON versions 
7.7.2: neuron.h.ri(0)=0.0003819718634205488 vs. neuron.h.ri(0, sec=parentsec)=1e+30
7.8.1: neuron.h.ri(0)=1+30 vs. neuron.h.ri(0, sec=parentsec)=1e+30
if I interupt test `test_cell_get_axial_currents_from_vmem_00`

**Comment by torbjone** [comment]
I have the impression that there is a problem with "cleaning up" as well here? For example, if I run a test: 
py.test -v LFPy/test/test_pointprocess.py
the first I get this
LFPy/test/test_pointprocess.py::testPointProcess::test_PointProcess_00 PASSED                                                                                                                           
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_00 PASSED                                                                                                                                    
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_01 Fatal Python error: Segmentation fault

but if I remove "test_Synapse_00", then I get:
LFPy/test/test_pointprocess.py::testPointProcess::test_PointProcess_00 PASSED                                                                                                                           
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_01 PASSED                                                                                                                                     
LFPy/test/test_pointprocess.py::testSynapse::test_Synapse_02 Fatal Python error: Segmentation fault

that is, "test_Synapse_01" seems to pass if it is ran first. Do you have this problem?

**Comment by espenhgn** [comment]
Yes, cleaning up is a problem. 

But I don't understand how succeeding tests may fail as `neuron.h('forall delete_section()')` should usually be executed before reading in morphology info in that case. 
 
I'm recompiling NEURON from source now, and will test more as soon as that works. I would just test whether or not the PyPI build of NEURON is the culprit. 

Edit: never mind, the test I suggested earlier may fail also with older versions.

**Comment by espenhgn** [comment]
More bizarre behaviour: Add 
```
        cell = None
        syn = None
        neuron.h('forall delete_section()')
        raise Exception
```
to the end of `test_Synapse_00` and it will fail obviously, but next test `test_Synapse_01` will succeed. 
Then add to `test_Synapse_01`:
```
        cell = None
        syn0 = None
        syn1 = None
        syn2 = None
        syn3 = None
        neuron.h('forall delete_section()')
        raise Exception
```
The next test `test_Synapse_02` will run and so forth. 

Remove the exceptions then `test_Synapse_01` will segfault. WTF. 

Here I execute just those tests by issuing `py.test -v LFPy/test/test_pointprocess.py::testSynapse`

**Comment by espenhgn** [comment]
I think we must be doing something utterly stupid somewhere in the Cell/and or PointProcess/Synapse class. The following test file fails on my end both with 7.7.2 and 7.8.1, espenhgn@iaxial and LFPy@master branch:
```
### test_neuron.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import unittest
import LFPy
import neuron

class testNeuron(unittest.TestCase):
    """
    test class
    """
    def test_neuron_00(self):
        cell = LFPy.Cell(morphology=os.path.join(LFPy.__path__[0], 'test', 'ball_and_sticks.hoc'))
        syn = LFPy.Synapse(cell=cell, idx=0, syntype='ExpSyn', weight=1.)
        del cell, syn
        neuron.h('forall delete_section()')

    def test_neuron_01(self):
        cell = LFPy.Cell(morphology=os.path.join(LFPy.__path__[0], 'test', 'ball_and_sticks.hoc'))
        syn = LFPy.Synapse(cell=cell, idx=0, syntype='ExpSyn', weight=1.)
        del cell, syn
        neuron.h('forall delete_section()')



if __name__ == '__main__':
    def test_neuron_02():
        cell = LFPy.Cell(morphology=os.path.join(LFPy.__path__[0], 'test', 'ball_and_sticks.hoc'))
        syn = LFPy.Synapse(cell=cell, idx=0, syntype='ExpSyn', weight=1.)
        del cell, syn
        neuron.h('forall delete_section()')

    test_neuron_02()
    test_neuron_02()
```

Running the file also causes a segfault.

**Comment by espenhgn** [comment]
Seems like segfaults in tests can be fixed by ending test function by `cell.strip_hoc_objects` to allow proper garbage collection. Been pushing some changes, not done.

**Comment by espenhgn** [comment]
Ok, the remaining failing tests w. 7.8.1 on my branch espenhgn@iaxial are all related to calculating axial currents from vmem. The rest could be fixed by calling `LFPy.Cell.strip_hoc_objects()` (which could be replaced by a `destroy` method).

**Comment by torbjone** [comment]
With that branch I also only seem to get errors from the axial currents from vmem, however, the example "example_MEA.py" does not run, with this error:

NEURON: Arg out of range in user function
 near line 0
 ^
        arc3d(2)
Traceback (most recent call last):
  File "examples/example_MEA.py", line 216, in <module>
    cell = LFPy.Cell(**cell_parameters)
  File "/home/tone/work/LFPy_debug_env/LFPy/LFPy/cell.py", line 285, in __init__
    self._collect_geometry()
  File "/home/tone/work/LFPy_debug_env/LFPy/LFPy/cell.py", line 668, in _collect_geometry
    _collect_geometry_neuron(self)
  File "/home/tone/work/LFPy_debug_env/LFPy/LFPy/run_simulation.py", line 321, in _collect_geometry_neuron
    L[i] = neuron.h.arc3d(i, sec=sec)
RuntimeError: hoc error

Does this happen at your side? Seems related to the "sec=sec" argument?

**Comment by espenhgn** [comment]
Just fixed it (forgot to add `sec=sec` to `neuron.h.n3d()` call)

**Comment by torbjone** [comment]
Thank you, that seems to work.

**Comment by espenhgn** [comment]
For future reference, this is the suggested way to record axial currents which can be applied during runtime:
https://www.neuron.yale.edu/phpBB2/viewtopic.php?t=238&highlight=axial+current

**Comment by espenhgn** [comment]
Closing, merged #210. Seems like all issues with NEURON>7.7.2 have been fixed.
